### PR TITLE
fix(config_setup): stop using regex to replace variables in scylla.yml

### PR DESCRIPTION
**Pull request by fruch**
since we move to use `scylla-machine-image` the scylla.yml in AMIs is
no longer the orignal one (i.e. it the same but without comments)
so there is not point of using regexs to try to preserve it.
this fix read the yaml with pyyaml, replaces what's needed and build it back.

Trello: https://trello.com/c/jqHnwCtU

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1575,173 +1575,100 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
                       timeout=300, throw_exc=True, src=yaml_file, dst=yaml_dst_path)
 
         with open(yaml_dst_path, 'r') as scylla_yaml_file:
-            scylla_yaml_contents = scylla_yaml_file.read()
+            # scylla_yaml_contents = scylla_yaml_file.read()
```
Comment (fgelcer) [inline sdcm/cluster.py]: remove dead code

**Comment by bentsi** [comment]
please add a link to the task in Trello

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1575,173 +1575,99 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
...
         else:
-            if 'auto_bootstrap' in scylla_yaml_contents:
-                if re.findall("auto_bootstrap: True", scylla_yaml_contents):
-                    self.log.debug('auto_bootstrap is not set as expected, update it to `False`.')
-                pattern = re.compile('auto_bootstrap:.*')
-                scylla_yaml_contents = pattern.sub('auto_bootstrap: False',
-                                                   scylla_yaml_contents)
-            else:
-                self.log.debug('auto_bootstrap is missing, set it `False`.')
-                scylla_yaml_contents += "\nauto_bootstrap: False\n"
```
Comment (roydahan) [inline sdcm/cluster.py:1698]: @fruch how from all this logic you decided that this if,else is enough?
Apparently, it's not.

This is what I see in the log of one cluster with an issue:
```
Node longevity-lwt-3h-4-0-db-node-5cfd2305-1 [18.207.221.117 | 172.30.0.162] (seed: False): Scylla YAML configuration:
api_address: 127.0.0.1
api_doc_dir: /opt/scylladb/api/api-doc/
api_port: 10000
api_ui_dir: /opt/scylladb/swagger-ui/dist/
auto_bootstrap: false
batch_size_fail_threshold_in_kb: 50
batch_size_warn_threshold_in_kb: 5
broadcast_rpc_address: 172.30.0.162
cas_contention_timeout_in_ms: 1000
client_encryption_options:
  enabled: false
cluster_name: longevity-lwt-3h-4-0-db-cluster-5cfd2305
commitlog_segment_size_in_mb: 32
commitlog_sync: periodic
commitlog_sync_period_in_ms: 10000
commitlog_total_space_in_mb: -1
endpoint_snitch: org.apache.cassandra.locator.Ec2Snitch
experimental: true
listen_address: 172.30.0.162
murmur3_partitioner_ignore_msb_bits: 12
native_transport_port: 9042
num_tokens: 256
partitioner: org.apache.cassandra.dht.Murmur3Partitioner
read_request_timeout_in_ms: 5000
rpc_address: 172.30.0.162
rpc_port: 9160
seed_provider:
- class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - seeds: 172.30.0.162
write_request_timeout_in_ms: 2000

```
If you see here, somehow the auto_bootstrap is set to "false" (lower case) when the cluster starts.
(We don't need to set anything at all, but we should start with the default scylla.yaml which now doesn't even have this parameter by default).

Then, in many place we are looking to "sed" False with True, but probably it doesn't do anything now.

I don't know where the lowercase "false" is coming from, but somehow changing the above logic caused issue.
we need to fix it ASAP.

@bentsi FYI

Comment (bentsi) [reply sdcm/cluster.py:1698]: created an urgent task task: https://trello.com/c/Djp6Oht5/1844-fix-autobootstrap-logic

Comment (fruch) [reply sdcm/cluster.py:1698]: YAML true, is always lower case.

where is the code that sed for "False", it should be replaced, reading correctly the yaml... or using this function

Comment (roydahan) [reply sdcm/cluster.py:1698]: thanks, it's related to scylla issue https://github.com/scylladb/scylla/issues/6120 and https://github.com/scylladb/scylla/issues/6121

Comment (fruch) [reply sdcm/cluster.py:1698]: this should fix it: 
https://github.com/scylladb/scylla-cluster-tests/pull/2031

**Pull request merged**